OpenRA.Game/Server/Server.cs

```diff
@@ -783,6 +848,12 @@ public void ReceiveOrders(Connection conn, int frame, byte[] data)
 		{
 			if (frame == 0)
 				InterpretServerOrders(conn, data);
+
+			// HACK: This is not a good way to detect and relay sync hashes but it is currently the only way
+			else if (conn != null && data.Length != 0 && (OrderType)data[0] == OrderType.SyncHash)
+				DispatchOrdersToClients(conn, frame, data, true);
+			else if (conn != null && LobbyInfo.GlobalSettings.UseNewNetcode && State == ServerState.GameStarted && GameState == GameState.Playing)
```

Comment:
We should throw if `conn == null` here, same as above `DispatchOrdersToClients` in the else branch would crash anyway. (I.e. if it is realistic that conn is null here, think about if we can or should gracefully ignore that and/or drop the null checks.)

Reply:
It doesn't really seem like `conn` can be null in the old code, so I'd just drop the checks.
```suggestion
			else if (data.Length != 0 && (OrderType)data[0] == OrderType.SyncHash)
				DispatchOrdersToClients(conn, frame, data, true);
			else if (LobbyInfo.GlobalSettings.UseNewNetcode && State == ServerState.GameStarted && GameState == GameState.Playing)
```